web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -95,18 +151,100 @@ export class DataStoreService {
    * @param data the source data in a dictionary keyed by string.
    */
   private static toLayer(id: string, data: DocumentData): Layer {
-    return new Layer(id, StringMap(data.name));
+    return new Layer(
+      id,
+      StringMap(data.name),
+      Map<string, Form>(
+        Object.keys(data.forms).map((id: string) => [
+          id as string,
+          DataStoreService.toForm(id, data.forms[id]),
+        ])
+      )
+    );
+  }
+
+  /**
+   * Converts the raw object representation deserialized from Firebase into an
+   * immutable Form instance.
+   *
+   * @param id the uuid of the form instance.
+   * @param data the source data in a dictionary keyed by string.
+   */
+  private static toForm(id: string, data: DocumentData): Form {
+    return new Form(
+      id,
+      Map<string, Field>(
+        Object.keys(data.elements).map((id: string) => [
+          id as string,
+          DataStoreService.toField(id, data.elements[id]),
+        ])
+      )
+    );
   }
 
   /**
-   * Converts a map of id to raw object deserialized from Firebase into an array
-   * of immutable Layer instances.
+   * Converts the raw object representation deserialized from Firebase into an
+   * immutable Field instance.
+   *
+   * @param id the uuid of the project instance.
+   * @param data the source data in a dictionary keyed by string.
+   * {
+   *   index: 0,
+   *   label: { 'en': 'Question 1' },
+   *   required: true,
+   *   type: 'text_field'
+   * }
+   * or
+   * {
+   *   index: 1,
+   *   label: { 'en': 'Question 2' },
+   *   required: false,
+   *   type: 'multiple_choice',
+   *   cardinality: 'select_one',
+   *   options: {
+   *     option001: {
+   *       index: 0,
+   *       code: 'A',
+   *       label: { 'en': 'Option A' }
+   *     },
+   *     // ...
+   *   }
+   */
+  private static toField(id: string, data: DocumentData): Field {
+    return new Field(
+      id,
+      FieldType.TEXT,
+      StringMap(data.label),
+      data.required,
+      data.options ? new MultipleChoice(
```

Comment:
I think you can use `data.options && new MultipleChoice(..)` with the ternary operator.